lib/log.js

```diff
@@ -175,105 +169,16 @@ Lug.prototype.notifyAttachedServices = function (name, request, data) {
 // These events indicate key points at which a particular
 // user has interacted with the service.
 
-Lug.prototype.activityEvent = function (event, request, data) {
-  if (! data || ! data.uid) {
-    this.error({ op: 'log.activityEvent', data: data })
-    return
-  }
-
-  var info = {
-    event: event
-  }
-
-  if (request.headers['user-agent']) {
-    info.userAgent = request.headers['user-agent']
-  }
-
-  optionallySetService(info, request)
-
-  Object.keys(data).forEach(function (key) {
-    info[key] = data[key]
-  })
-
-  this.logger.info('activityEvent', info)
-  this.statsd.write(info)
+Lug.prototype.activityEvent = function (data) {
+  this.logger.info('activityEvent', data)
```

Comment:
You were dead right about this, of course. I've moved the validation back into the log methods and added appropriate tests.

I'll carry your r+ forward and merge this when its green.